### DISCUSSION TITLE
embed-message

### DISCUSSION THREAD

**Discussion by giscus** [brainboxdotcc/dpp-comments#31] [General]
# embed-message

A lightweight C++ Discord API library supporting the entire Discord API, including Slash Commands, Voice/Audio, Sharding, Clustering and more!

https://dpp.dev/embed-message.html

**Comment by GiovanniRaseraF** [comment]
I think there is a typo
this:
```c++
if (event.command.get_command_name == "embed") {
```
should be this:
```c++
if (event.command.get_command_name() == "embed") {
```

**Reply by Jaskowicz1** [reply]
Yep, that's a typo! Submitting a PR for this :)

**Reply by GiovanniRaseraF** [reply]
Can i try to do it ? :)

**Reply by Jaskowicz1** [reply]
Yep, just created my PR and saw this, gonna close my PR so you can do it :)

**Comment by KatCote** [comment]
```
dpp::i_message_content
```

this causes such an error:
```
[2024-03-12 16:30:29] DEBUG: Connecting new session...
[2024-03-12 16:30:29] WARN: OOF! Error from underlying websocket: 4014: Disallowed intent(s)
[2024-03-12 16:30:29] DEBUG: Attempting reconnection of shard 0 to wss://gateway.discord.gg
```

**Reply by Jaskowicz1** [reply]
You haven't turned on the intent in your bot's dashboard on the discord developers site!

Turn that on and restart your bot, it should work :)

**Reply by KatCote** [reply]
Oh thanks

**Reply by HunterZ** [reply]
Can you explain why the intent is needed for this example to work? Can a warning be added to the article so that people don't trip over this?

**Reply by ruslan-ilesik** [reply]
As far as i can see there is no need for message content intent, will double check and probably fix it

**Reply by Jaskowicz1** [reply]
> Can you explain why the intent is needed for this example to work? Can a warning be added to the article so that people don't trip over this?

It isn't needed, it should be removed tbh.

**Reply by ruslan-ilesik** [reply]
working on it

**Reply by ruslan-ilesik** [reply]
> working on it

https://github.com/brainboxdotcc/DPP/pull/1250

**Comment by Shmurkio** [comment]
It doesn't work for me. I think I set up everything correctly but when I compile and run it it doesn't print anything in the console and my bot is not started either.

**Reply by Jaskowicz1** [reply]
Hey! If you're able to join the Discord Server so we can get a bit more of an understanding, we can try help out there!

**Reply by HunterZ** [reply]
There's definitely something wrong with this example (besides the change in how the token parameter looks - I changed it to `BOT_TOKEN` to match the "Creating Your First Bot" example so that it would use my token - according to the debugger, it just silently hangs once `bot.start(dpp::st_wait);` is reached.

**Reply by Jaskowicz1** [reply]
> There's definitely something wrong with this example (besides the change in how the token parameter looks - I changed it to `BOT_TOKEN` to match the "Creating Your First Bot" example so that it would use my token - according to the debugger, it just silently hangs once `bot.start(dpp::st_wait);` is reached.

There shouldn't be, I've tested it and can confirm it works. Please join the Discord Server and give us more information :)

**Reply by ruslan-ilesik** [reply]
> There's definitely something wrong with this example (besides the change in how the token parameter looks - I changed it to `BOT_TOKEN` to match the "Creating Your First Bot" example so that it would use my token - according to the debugger, it just silently hangs once `bot.start(dpp::st_wait);` is reached.

Example did not had logger defined, as others examples do, fixed in same pr: https://github.com/brainboxdotcc/DPP/pull/1250

**Reply by HunterZ** [reply]
Yep, thanks, just figured this out by diffing the two examples - I think not having the logger was hiding the errant intent issue.
